### PR TITLE
Make status endpoint less fragile to timeouts

### PR DESCRIPTION
**Pull request by wouterdb**
# Description
Make status endpoint less fragile to timeouts
# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/server/server.py [hunk 1]
```diff
@@ -128,15 +129,21 @@ async def get_server_status(self) -> StatusResponse:
...
             )
 
-        slices = []
-        for slice_name, slice in self._server.get_slices().items():
+        async def collect_for_slice(slice_name: str, slice: protocol.ServerSlice) -> SliceStatus:
             try:
-                slices.append(SliceStatus(name=slice_name, status=await slice.get_status()))
+                return SliceStatus(name=slice_name, status=await asyncio.wait_for(slice.get_status(), 0.1))
+            except TimeoutError:
+                return SliceStatus(name=slice_name, status={"error": "timeout after 100ms"})
```
Comment (sanderr) [inline src/inmanta/server/server.py:136]: Should we have the frontend retry regularly if this occurs? Also for handling these new error responses in general, should we create a frontend ticket?

**Review by sanderr — APPROVED** [review]

File: src/inmanta/server/server.py (hunk 1 above)
Comment (wouterdb) [reply src/inmanta/server/server.py:136]:
```suggestion
                return SliceStatus(name=slice_name, status={"error": f"timeout on data collection for {slice_name}, consult the server log for additional information"})
```

Comment (wouterdb) [reply src/inmanta/server/server.py]: I discussed it and this is what we came up with

Comment (sanderr) [reply src/inmanta/server/server.py]: So no retries? I assume this is an exceptional case and indicates something is wrong?

Comment (wouterdb) [reply src/inmanta/server/server.py]: Indeed. (and they naturally retry every 5 seconds)

File: src/inmanta/server/services/databaseservice.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ def start_monitor(self) -> None:
     async def get_connection_status(self) -> bool:
         if self._pool is not None and not self._pool._closing and not self._pool._closed:
             try:
-                async with self._pool.acquire(timeout=10):
+                async with self._pool.acquire(timeout=5):
```
Comment (Hugo-Inmanta) [inline src/inmanta/server/services/databaseservice.py:119]: I'm curious, what's the rationale behind this change ?

**Review by Hugo-Inmanta — APPROVED** [review]

File: src/inmanta/server/services/databaseservice.py (hunk 2 above)
Comment (wouterdb) [reply src/inmanta/server/services/databaseservice.py:119]: oh, right not required, thx

Comment (wouterdb) [inline src/inmanta/server/services/databaseservice.py:119]:
```suggestion
                async with self._pool.acquire(timeout=10):
```
